sklearn/utils/estimator_checks.py

```diff
@@ -1360,6 +1390,62 @@ def check_classifiers_train(name, classifier_orig):
                 assert_array_equal(np.argsort(y_log_prob), np.argsort(y_prob))
 
 
+def check_outliers_train(name, estimator_orig):
+    X, _ = make_blobs(n_samples=300, random_state=0)
+    X = shuffle(X, random_state=7)
+    n_samples, n_features = X.shape
+    estimator = clone(estimator_orig)
+    set_random_state(estimator)
+
+    # fit
+    estimator.fit(X)
+    # with lists
+    estimator.fit(X.tolist())
```

Comment:
ideally we check that this makes the *same* predictions as fitting with an array. We should possibly have a similar check to `check_estimators_dtypes` that checks *all* estimators with lists, but it looks like we don't.

Reply:
This is how it is done in `check_classifier_train` but yes we should maybe open a specific PR for that.